### DISCUSSION TITLE
Support Vue 2.7

### DISCUSSION THREAD

**Discussion by nolde** [chakra-ui/zag#859] [General]
# 🚀 Feature request

Vue 2.7 has native support to Compositions API. However, [@zag-js/vue](https://github.com/chakra-ui/zag/blob/main/packages/frameworks/vue/package.json) determines only Vue 3 as peer dependency. **Would it be possible to move it to Vue 2.7** ?

## 🧱 Problem Statement / Justification

 Checking your usage, Vue 2.7 and above would be supported just fine. Some systems are not ready to move to Vue 3 for a myriad of reasons, but could still benefit from using zag.

## ✅ Proposed solution or API

```
"peerDependencies": {
    "vue": ">=2.7"
},
```

**Comment by segunadebayo** [comment]
Thanks for bringing this up @nolde

We don't have plans to support Vue versions less than 3.0 due to the updated typings we get from Vue.

An alternative would be to use pnpm patch or similar to tweak the package.json and see if it still works.

**Comment by segunadebayo** [comment]
Closing — Vue 2 reached end-of-life in December 2023 and is no longer supported.
